src/Data/Fin/Permutation/Transposition/List.agda

```diff
@@ -29,43 +31,68 @@ private
 
 -- This decomposition is not a unique representation of the original
 -- permutation but can be used to simply proofs about permutations (by
--- instead inducting on the list of transpositions).
+-- instead inducting on the list of transpositions, where a
+-- transposition is a permutation swapping two distinct elements and
+-- leaving the rest in place).
 
 TranspositionList : ℕ → Set
-TranspositionList n = List (Fin n × Fin n)
+TranspositionList n = List (∃₂ λ (i j : Fin n) → i ≢ j)
```

Comment:
Hmm I'm sorry to reopen the discussion, but looking at this with a fresh pair of eyes, wouldn't a more modular and reusable solution be to keep the definition as is and then define a predicate:
```
NonTrivialEntries : TranspositionList n → Set
NonTrivialEntries xs = All (curry _≢_) xs
```
which could then be taken as an argument in the proofs that need it. The advantages of this would be:

1. Preserves backwards compatibility
2. Provides more choice to the user - I can imagine a scenario where the reflexive entries do hold semantic information.
3. Allows us to reuse lemmas from `Data.List.Relation.Unary.All`

I'm open to suggestions about the name `NonTrivialEntries`?

Reply:
I really want to argue against this, but the truth is that for everything currently existing, that the transposition list has no trivial entries is completely irrelevant. I think the right thing to do for now might be to not talk about that constraint until we've got something that actually uses it.

Reply:
Yup, happy with that solution!

Reply:
So what needs to be done to implement that solution?